visualization/app/codeCharta/ui/copyToClipboardButton/util/getFilenamesWithHighestMetrics.ts

```diff
@@ -0,0 +1,70 @@
+import { hierarchy } from "d3-hierarchy"
+import { CodeMapNode, NodeType } from "../../../codeCharta.model"
+
+export type FileToValue = { name: string; value: number }
+
+const MAX_ENTRIES = 10
+
+export function getFilenamesWithHighestMetrics(node: CodeMapNode) {
+	const fileToValueByAttributes = new Map<string, FileToValue[]>()
+	for (const { data } of hierarchy(node)) {
+		if (data.type === NodeType.FILE && data.attributes) {
+			for (const [key, value] of Object.entries(data.attributes)) {
+				maybeAddToMap(key, value, data.name, fileToValueByAttributes)
+			}
+		}
+	}
+	return fileToValueByAttributes
+}
+
+export function maybeAddToMap(key: string, value: number, fileName: string, map: Map<string, FileToValue[]>) {
+	const keyIsNew = !map.has(key)
+	if (keyIsNew) {
+		map.set(key, [{ name: fileName, value }])
+		return
+	}
+
+	const newPair = { name: fileName, value }
+	const previousValues = map.get(key)
+	if (previousValues.length < MAX_ENTRIES) {
+		insertSorted(previousValues, newPair)
+		return
+	}
+
+	insertInFullDescendingList(newPair, previousValues)
```

Comment:
```suggestion
	insertSorted(previousValues, newPair)
	map.set(key, previousValues.slice(0, MAX_ENTRIES
```

You also could do this. You have a more maintainable and readable code and only use on insert method. This implementation is approx. 10 to 15 seconds slower than the current version.